geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/planning/QueryInterceptor.scala

```diff
@@ -40,6 +46,14 @@ trait QueryInterceptor extends Closeable {
     * @param query query
     */
   def rewrite(query: Query): Unit
+
+  /**
+   * Hook to allow interception of a query after extracting the query values
+   *
+   * @param strategy query strategy
+   * @return an exception if the query should be stopped
+   */
+  def guard(strategy: QueryStrategy): Option[IllegalArgumentException] = None
```

Comment:
If guard has a default implementation, should rewrite have one as well?

Or is the default implementation added to improve compatibility with existing implementations of the QueryInterceptor interface?

Reply:
right, just for compatibility